ironfish-cli/src/commands/wallet/send.ts

```diff
@@ -143,56 +156,120 @@ export class Send extends IronfishCommand {
       this.exit(1)
     }
 
-    if (fee == null) {
-      let suggestedFee = ''
-      try {
-        const response = await client.estimateFee({
-          fromAccountName: from,
+    if (fee == null && feeRate == null) {
+      const feeRates = await client.estimateFeeRates()
+      const feeRateOptions = []
+
+      if (feeRates.content.low) {
+        const createResponse = await client.createTransaction({
+          sender: from,
           receives: [
             {
               publicAddress: to,
               amount: CurrencyUtils.encode(amount),
-              memo: memo,
+              memo,
+              assetId,
             },
           ],
+          feeRate: feeRates.content.low,
+          expiration: expiration,
         })
 
-        switch (flags.priority) {
-          case 'low':
-            suggestedFee = CurrencyUtils.renderIron(response.content.low)
-            break
-          case 'high':
-            suggestedFee = CurrencyUtils.renderIron(response.content.high)
-            break
-          default:
-            suggestedFee = CurrencyUtils.renderIron(response.content.medium)
-        }
-      } catch {
-        suggestedFee = ''
+        const rawTransactionBytes = Buffer.from(createResponse.content.transaction, 'hex')
+        const rawTransaction = RawTransactionSerde.deserialize(rawTransactionBytes)
+
+        feeRateOptions.push({
+          value: feeRates.content.low,
+          name: `Low: ${rawTransaction.fee} IRON`,
+        })
       }
 
-      const input = await CliUx.ux.prompt(
-        `Enter the fee amount in $IRON (min: ${CurrencyUtils.renderIron(
-          1n,
-        )} recommended: ${suggestedFee})`,
+      if (feeRates.content.medium !== feeRates.content.low) {
+        const createResponse = await client.createTransaction({
+          sender: from,
+          receives: [
+            {
+              publicAddress: to,
+              amount: CurrencyUtils.encode(amount),
+              memo,
+              assetId,
+            },
+          ],
+          feeRate: feeRates.content.medium,
+          expiration: expiration,
+        })
```

Comment:
Talked offline about reusing the repeated fields in the request object and using `Promise.all` on the multiple `createTransaction` calls

Reply:
updated